facia-tool/public/js/widgets/columns/fronts-search.js

```diff
@@ -0,0 +1,81 @@
+import ko from 'knockout';
+import _ from 'underscore';
+import ColumnWidget from 'widgets/column-widget';
+import CONST from 'constants/defaults';
+import generateCollections from 'utils/generate-collections';
+
+export default class SearchConfig extends ColumnWidget {
+    constructor(params, element) {
+        super(params, element);
+        this.searchTerm = ko.observable('');
+        this.searchedFronts = ko.observableArray([]);
+
+        var self = this;
+        var originalFronts;
+        populateOriginalFrontList();
+        this.subscribeOn(this.baseModel.state, populateOriginalFrontList);
+
+        function populateOriginalFrontList() {
+            originalFronts =  _.reduce(self.baseModel.frontsList(), function (frontList, front) {
+
+                if (_.every(CONST.askForConfirmation, function (element) {
+                    return front.id !== element;
+                }) ) {
+
+                    let frontToSearch = { id: front.id.toLowerCase() };
+
+                    frontToSearch.collections = generateCollections(front.collections);
+                    frontList.push(frontToSearch);
+                }
+                return frontList;
+            }, []);
+        }
+
+        this.searchTerm.subscribe(function (newTerms) {
```

Comment:
it is safe to subscribe to observables in the same class, but I still prefer to use `subscribeOn`, for consistency.

Also it would be nice to put the body of this function in its own method, say `search`

```js
class {
   constructor() {}
   search() {}
}
```